### PR TITLE
Integrate "Target-specific code" (ocsigen/js_of_ocaml#1655)

### PR DESCRIPTION
**Pull request by OlivierNicole**
This integrates PRs ocsigen/js_of_ocaml#1655 and its small preliminary PR ocsigen/js_of_ocaml#1692. The two subsequent commits make the necessary changes in the code of wasm_of_ocaml.

### CODE REVIEW

File: dune-workspace.dev [hunk 1]
```diff
@@ -1,4 +1,4 @@
-(lang dune 3.7)
+(lang dune 3.15)
```
Comment (vouillon) [inline dune-workspace.dev:1]: Maybe we should put 3.17 everywhere.
```suggestion
(lang dune 3.17)
```

Comment (OlivierNicole) [reply dune-workspace.dev]: Done.

File: compiler/bin-wasm_of_ocaml/compile.ml [hunk 2]
```diff
@@ -270,15 +272,8 @@ let run
   List.iter builtin ~f:(fun t ->
       let filename = Builtins.File.name t in
       let runtimes = Linker.Fragment.parse_builtin t in
-      Linker.load_fragments
-        ~ignore_always_annotation:true
```
Comment (vouillon) [inline compiler/bin-wasm_of_ocaml/compile.ml:274]: To remove this argument, we need https://github.com/ocsigen/js_of_ocaml/pull/1674/

**Comment by vouillon** [comment]
The type of `Driver.optimize` is still different between Js_of_ocaml and Wasm_of_ocaml.

**Comment by vouillon** [comment]
`Parse_bytecode.from_exe` should not longer have a `target` parameter.

**Comment by OlivierNicole** [comment]
That should be fixed.

**Pull request merged**